llvm/utils/TableGen/Common/CodeGenTarget.cpp

```diff
@@ -324,9 +325,8 @@ void CodeGenTarget::ComputeInstrsByEnum() const {
       });
 
   // Assign an enum value to each instruction according to the sorted order.
-  unsigned Num = 0;
-  for (const CodeGenInstruction *Inst : InstrsByEnum)
-    Inst->EnumVal = Num++;
+  for (const auto &[Idx, Inst] : enumerate(InstrsByEnum))
+    const_cast<CodeGenInstruction *>(Inst)->EnumVal = Idx;
```

Comment:
`mutable` was invented to avoid exactly this

Reply:
Right, though mutable is universally true (i.e., any other code can change it without a const_cast) whereas with this you have to do something intentional to mutate the var, so safer. But I am fine going back to mutable.

Reply:
Though, there's a lot of mutable variables TableGen/Common/CodeGen* files, so makes sense to go back to mutable for consistency :) I'll do that

Reply:
Done.